storage/innobase/handler/ha_innodb.cc

```diff
@@ -15296,8 +15296,11 @@ be dropped
 @return	error number
 @retval 0 on success */
 int ha_innobase::delete_table(const char *name, const dd::Table *table_def) {
+  // during ddse change, s_dd_table_ids may contain old dd table ids, thus
+  // allow drop/rename if current se isn't target ddse
   if (table_def != nullptr &&
-      dict_sys_t::is_dd_table_id(table_def->se_private_id())) {
+      dict_sys_t::is_dd_table_id(table_def->se_private_id()) &&
+      (default_dd_storage_engine == DEFAULT_DD_INNODB)) {
```

Comment:
This does not look right unless I'm missing something. It looks like it will allow user `DROP TABLE mysql.tables` when InnoDB is the DDSE. Can the DDSE change check be added here?

https://github.com/facebook/mysql-5.6/pull/1367 adds `innobase_is_ddse` helper, which should be used here if that PR lands first, otherwise I'll clean up in one of the later PRs.

Reply:
how about add a DD system thread check.. Currently DD_bootstrap_ctx isn't exposed to SE and is_dd_system_thread() is widely used in DD layer..

Reply:
Let's use `is_dd_system_thread`